.github/workflows/branch.yml

```diff
@@ -0,0 +1,44 @@
+---
+
+#########################
+#########################
+## Deploy Branch Docker Image to ECR ##
+#########################
+#########################
+
+
+name: Branch dotnet image push to ECR
+on:
+  push:
+    branches-ignore:
+      - master
+jobs:
+  build:
+    name: Deploy Branch Docker Image
+    runs-on: [test]
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Get current date
+        run: echo "BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ')" >> ${GITHUB_ENV}
+
+      - name: Login to ECR
+        env:
+          ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
+          ECR_REGISTRY: ${{ secrets.ECR_REGISTRY }}
+        run: docker login -u AWS -p $(aws ecr-public get-login-password --region us-east-1) $ECR_REGISTRY
+
+      - name: Build and push
+        env:
+          ECR_REGISTRY: ${{ secrets.ECR_REGISTRY }}
+          ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
+        run: |
+          docker build \
+            --build-arg BUILD_DATE \
+            --build-arg BUILD_REVISION=$GITHUB_SHA \
+            --build-arg BUILD_VERSION=$GITHUB_SHA \
+            -t $ECR_REGISTRY/$ECR_REPOSITORY:sha-${GITHUB_SHA:0:7} .
```

Comment:
Please use lazy action setup and get the image version from there

Reply:
Removed trimmed github sha and taking raw github sha as image tag.